src/clojure/tdlib_json/core.clj

```diff
@@ -1,48 +1,43 @@
 (ns tdlib-json.core
-  (:require [cheshire.core :as che]
-            [clojure.core.async :as async])
-  (:import [tdlib_json TgClient]))
+  (:require [clojure.core.async :as async])
+  (:import [tdlib_json TgJsonClient]))
 
-(def client (atom nil))
 
-(defn create-client [path-to-lib timeout]
-  (TgClient. path-to-lib timeout))
+(def client (atom nil))
 
 (def message-queue (atom nil))
 
-(defn jsonify [messageMap] (che/generate-string messageMap))
 
-(defn json-parse [string]
-  (che/parse-string string true))
+(defn create-client [path-to-lib verbosity]
+  (TgJsonClient. path-to-lib verbosity))
+
+(defn client-send [message]
+  (.send ^TgJsonClient @client message))
 
-(defn client-execute
-  ([messageMap] (. @client execute (jsonify messageMap))))
+(defn client-execute [message]
+  (.execute ^TgJsonClient @client message))
 
-(defn client-receive
-  []
-  (json-parse (. @client receive)))
+(defn client-receive [timeout]
+  (.receive ^TgJsonClient @client timeout))
 
 (defn client-destroy []
+  (client-send "{\"@type\":\"close\"}")
+  (.destroy ^TgJsonClient @client)
```

Comment:
There will be no delay between these two calls. On close client sends updateAuthorizationState with authorizationStateClosed message. After this it is safe to destroy client. I am going to add this.